hsapp/app.go

```diff
@@ -222,22 +223,55 @@ func (a *App) render() {
 	a.TextureLoader.ResumeLoadingTextures()
 }
 
+// please note, that this steps will not affect app language
+// it will only load an appropriate glyph ranges for
+// displayed text (e.g. for string/font table editors)
 func (a *App) setupFonts() {
-	// Note: To support other languages we'll have to do something with glyph ranges here...
-	// ranges := imgui.NewGlyphRanges()
-	// rb := imgui.NewFontGlyphRangesBuilder()
-	// rb.AddRanges(imgui.CurrentIO().Fonts().GlyphRangesJapanese())
-	// rb.AddRanges(imgui.CurrentIO().Fonts().GlyphRangesChineseSimplifiedCommon())
-	// rb.AddRanges(imgui.CurrentIO().Fonts().GlyphRangesCyrillic())
-	// rb.AddRanges(imgui.CurrentIO().Fonts().GlyphRangesKorean())
-	// rb.BuildRanges(ranges)
-	// imgui.CurrentIO().Fonts().AddFontFromFileTTFV("NotoSans-Regular.ttf", 17, 0, imgui.CurrentIO().Fonts().GlyphRangesJapanese())
-	imgui.CurrentIO().Fonts().AddFontFromMemoryTTF(hsassets.FontNotoSansRegular, 17)
-	a.fontFixed = imgui.CurrentIO().Fonts().AddFontFromMemoryTTF(hsassets.FontCascadiaCode, 15)
-	a.fontFixedSmall = imgui.CurrentIO().Fonts().AddFontFromMemoryTTF(hsassets.FontCascadiaCode, 12)
-	a.diabloRegularFont = imgui.CurrentIO().Fonts().AddFontFromMemoryTTF(hsassets.FontDiabloRegular, 15)
-	a.diabloBoldFont = imgui.CurrentIO().Fonts().AddFontFromMemoryTTF(hsassets.FontDiabloBold, 30)
-	imgui.CurrentStyle().ScaleAllSizes(1)
+	// get font manager
+	fonts := g.Context.IO().Fonts()
+
+	// create glyph ranges
+	ranges := imgui.NewGlyphRanges()
+
+	// create glyph ranges builder
+	builder := imgui.NewFontGlyphRangesBuilder()
+
+	// add default ranges
+	builder.AddRanges(fonts.GlyphRangesDefault())
+
+	var font []byte = hsassets.FontNotoSansRegular
+	// add special ranges
+	switch a.config.Locale {
+	// glyphs supported by default
+	case hsenum.LocaleEnglish, hsenum.LocaleGerman,
+		hsenum.LocaleFrench, hsenum.LocaleItalien,
+		hsenum.LocaleSpanish:
+		// noop
+	case hsenum.LocaleChinaTraditional:
+		font = hsassets.FontSourceHanSerif
+
+		builder.AddRanges(fonts.GlyphRangesChineseFull())
+	case hsenum.LocaleKorean:
+		font = hsassets.FontSourceHanSerif
+
+		builder.AddRanges(fonts.GlyphRangesKorean())
+	case hsenum.LocalePolish:
+		builder.AddText(hsenum.PolishSpecialCharacters)
+	}
+
+	// build ranges
+	builder.BuildRanges(ranges)
+
+	// setup default font
+	fonts.AddFontFromMemoryTTFV(font, 17, 0, ranges.Data())
+
+	// please note, that the following fonts will not use
+	// previously generated glyph ranges.
+	// they'll have a default range
+	a.fontFixed = fonts.AddFontFromMemoryTTF(hsassets.FontCascadiaCode, 15)
+	a.fontFixedSmall = fonts.AddFontFromMemoryTTF(hsassets.FontCascadiaCode, 12)
+	a.diabloRegularFont = fonts.AddFontFromMemoryTTF(hsassets.FontDiabloRegular, 15)
+	a.diabloBoldFont = fonts.AddFontFromMemoryTTF(hsassets.FontDiabloBold, 30)
```

Comment:
Should we put these font size integers into constants somewhere? Maybe not in this commit, but might be something nice to have in the config file.

Reply:
done in #281